app/Foliage/CmdBuild.hs

```diff
@@ -312,3 +315,20 @@ mkTarEntry contents indexFile timestamp =
 anchorPath :: Path Absolute -> (RepoLayout -> RepoPath) -> FilePath
 anchorPath outputDirRoot p =
   toFilePath $ anchorRepoPathLocally outputDirRoot $ p hackageRepoLayout
+
+makePreferedVersionsFiles :: FilePath -> [PreparedPackageVersion] -> Action ()
+makePreferedVersionsFiles outputDir packages = traced "writing prefered versions" $ do
```

Comment:
Those `makeAllPackage*` functions I call at the top are just to make the website, so don't take them as an example. The functions `prepareIndexPkgCabal` and `prepareIndexPkgMetadata` are more relevant.